File: MintPlayer.Spark.Abstractions/ProgramUnit.cs
namespace MintPlayer.Spark.Abstractions;

public sealed class ProgramUnitsConfiguration
{
    public ProgramUnitGroup[] ProgramUnitGroups { get; set; } = [];
}

public sealed class ProgramUnitGroup
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Icon { get; set; }
    public int Order { get; set; }
    public ProgramUnit[] ProgramUnits { get; set; } = [];
}

public sealed class ProgramUnit
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Icon { get; set; }
    public required string Type { get; set; }
    public Guid? QueryId { get; set; }
    public Guid? PersistentObjectId { get; set; }
    public int Order { get; set; }
}

File: Demo/DemoApp/DemoApp/ClientApp/src/app/core/models/program-unit.ts
export interface ProgramUnit {
  id: string;
  name: string;
  icon?: string;
  type: string;
  queryId?: string;
  persistentObjectId?: string;
  order: number;
}

export interface ProgramUnitGroup {
  id: string;
  name: string;
  icon?: string;
  order: number;
  programUnits: ProgramUnit[];
}

export interface ProgramUnitsConfiguration {
  programUnitGroups: ProgramUnitGroup[];
}
